courses/fundamentals_of_ada/adv_240_tasking.rst

```diff
@@ -625,6 +625,61 @@ Waiting On Different Entries
        exit;
      end select;
 
+------------------------
+Protected Object Entries
+------------------------
+
+* **Special** kind of protected :ada:`procedure`
+* May use a **barrier**, that **only** allows call on a **boolean** condition
+* Barrier is **evaluated** and may be **relieved** when
+
+   - A task calls :ada:`entry`
+   - A protected :ada:`entry` or :ada:`procedure` is **exited**
+
+* Several tasks can be waiting on the same :ada:`entry`
+
+    - Only **one** will be re-activated when the barrier is relieved
+
+.. code:: Ada
+
+   protected body Stack is
+      entry Push (V : Integer) when Size < Buffer'Length is
+      ...
+      entry Pop  (V : out Integer) when Size > 0 is
+      ...
+   end Object;
+
+-------------------
+Guard Expressions
```

Comment:
Ok, more confusion :(
A Guard Expression is exactly what we just called a "barrier" in the previous slide.
I would move this slide to before "Protected Object Entries", and then remove the reference to protected objects on this slide (as we'll discuss it in the following slide)

Reply:
Guard and barrier are actually two different concepts in the ARM, I tried clarifying that, WDYT

Reply:
OK, now _I_ understand it better :)